src/dinitctl.cc

```diff
@@ -193,6 +200,10 @@ int dinitctl_main(int argc, char **argv)
                     break;
                 }
             }
+            else if (command == command_t::SIG_SEND && i == 2
```

Comment:
What is the `i == 2` check for?

Reply:
For throw "unrecognised/invalid argument: --list" on `dinitctl signal <signal-name> --list` situation.

Reply:
Ok I guess you're trying to check that the command line is exactly "dinitctl signal --list". But:

- If I do "dinitctl signal KILL --list" -- It will say "--list" is an "unrecognized/invalid option" even though it is not. I just committed a fix for the same issue for other commands! I don't want to reintroduce the same bug now.
- what if I do: "dinitctl -u signal --list" for example? I don't think this should error (even though the `-u` doesn't really have any effect in this case).
- in general this is very brittle.

Reply:
> For throw "unrecognised/invalid argument: --list" on `dinitctl signal <signal-name> --list` situation.

Yeah but that's wrong. It's not an unrecognised/invalid argument. It's a recognized, valid argument that's given in the wrong context.

Reply:
I literally fixed this same issue just a few hours back:https://github.com/davmac314/dinit/commit/c3725a9e1dc8fe702efb737de6eeb0860516869f

I don't want to bring the same bug back now!